modules/oe_theme_content_event/src/Plugin/ExtraField/Display/ContactsExtraField.php

```diff
@@ -51,6 +51,14 @@ public function viewElements(ContentEntityInterface $entity) {
     foreach ($entity->get('oe_event_contact')->referencedEntities() as $contact) {
       $bundle = $contact->bundle();
 
+      if (!$contact->access('view', \Drupal::currentUser())) {
+        return [];
+      }
+
+      if (!$contact->isPublished()) {
+        continue;
+      }
+
```

Comment:
It's probably not needed. There is permission to view published and and view unpublished entities.
Check build/modules/contrib/oe_content/modules/oe_content_entity/src/PermissionCallbacksBase.php